File: core/src/main/java/com/victorlh/android/framework/core/errores/ErrorGeneral.java
package com.victorlh.android.framework.core.errores;

public class ErrorGeneral extends Exception {

	private IAvisoError[] avisos;

	public ErrorGeneral() {
	}

	public ErrorGeneral(String message) {
		super(message);
	}

	public ErrorGeneral(String message, Throwable cause) {
		super(message, cause);
	}

	public ErrorGeneral(Throwable cause) {
		super(cause);
	}

	protected void setAvisos(IAvisoError[] avisos) {
		this.avisos = avisos;
	}

	public IAvisoError[] getAvisos() {
		return avisos == null ? new IAvisoError[0] : avisos;
	}
}

File: core/src/main/java/com/victorlh/android/framework/core/errores/ErrorGeneral.kt
package com.victorlh.android.framework.core.errores

open class ErrorGeneral : Exception {

	constructor()
	constructor(message: String) : super(message)
	constructor(message: String, cause: Throwable) : super(message, cause)
	constructor(cause: Throwable) : super(cause)

	var avisos: Array<AvisoError> = emptyArray()

}
